src/bitcoind.cpp

```diff
@@ -191,5 +191,10 @@ int main(int argc, char* argv[])
     // Connect bitcoind signal handlers
     noui_connect();
 
-    return (AppInit(argc, argv) ? EXIT_SUCCESS : EXIT_FAILURE);
+    int status = AppInit(argc, argv);
+
+    if (!status) {
```

Comment:
i think if you do shutdown here it will only work for coind but not qt but in init.cpp it will for both? not entirely sure though